### PR TITLE
Format all code with `cargo fmt`

### PR DESCRIPTION
**Pull request by exrook**
To assuage the concerns noted in #2338 regarding blocking other in progress PRs I have developed a process to rebase branches on top of this branch by formatting each commit. I have already rebased the following PRs which have no merge conflicts with `master`.

https://github.com/exrook/futures-rs/compare/format...exrook:pr2384 2384
https://github.com/exrook/futures-rs/compare/format...exrook:pr2379 2379
https://github.com/exrook/futures-rs/compare/format...exrook:pr2374 2374
https://github.com/exrook/futures-rs/compare/format...exrook:pr2355 2355
https://github.com/exrook/futures-rs/compare/format...exrook:pr2352 2352
https://github.com/exrook/futures-rs/compare/format...exrook:pr2343 2343
https://github.com/exrook/futures-rs/compare/format...exrook:pr2342 2342
https://github.com/exrook/futures-rs/compare/format...exrook:pr2328 2328
https://github.com/exrook/futures-rs/compare/format...exrook:pr2244 2244
https://github.com/exrook/futures-rs/compare/format...exrook:pr1956 1956

Additionally I have created `.git-blame-ignore-revs` and added the format commit to it to improve `git-blame` output, see:
https://git-scm.com/docs/git-blame#Documentation/git-blame.txt---ignore-revs-fileltfilegt
```
git config blame.ignoreRevsFile .git-blame-ignore-revs
```

<details><summary>Instructions/script for updating a branch</summary>
<p>

## Convert current branch (manual, for branches with conflicts)

Do as `rebase` instructs to resolve any conflicts here!
```sh
git rebase master
```
Format all commits, should not require intervention
```sh
REV_SPEC=$(git rev-list master..HEAD | tail -n1)~~..HEAD
git filter-branch -f --tree-filter 'cargo fmt -- --config disable_all_formatting=false --config use_small_heuristics="Max"' $REV_SPEC
```
Rebase onto the formatted branch, should not require intervention
```sh
git fetch upstream pull/2385/head:format # grab the branch from this PR
git rebase format
```
Push rewritten commits
```sh
git push -f
```

## Batch convert PRs: usage `./script.sh <pr numbers>`
Will have problems with prs with merge conflicts. You probably want to do this on a fresh clone. Replace `upstream` with the name of your `rust-lang/futures-rs` remote
```bash
#!/bin/bash

export FILTER_BRANCH_SQUELCH_WARNING=1
PRS=$@
cd futures-rs
git checkout master
git fetch -f upstream pull/2385/head:format
for pull in $PRS; do
    echo PR $pull
    git fetch -f upstream pull/$pull/head:pr$pull
    echo FETCH DONE
    git checkout pr$pull
    echo CHECKOUT DONE
    git rebase master || git rebase --abort
    echo REBASE 1 DONE
    REV_SPEC=$(git rev-list origin/master..HEAD | tail -n1)~~..HEAD
    git filter-branch -f --tree-filter 'cargo fmt -- --config disable_all_formatting=false --config use_small_heuristics="Max"' $REV_SPEC && \
        echo FILTER BRANCH DONE && \
        git rebase format || git rebase --abort
        echo REBASE 2 DONE && \
        git push -fu origin pr$pull && \
        echo PUSH DONE
    echo PR $pull DONE
done
cd ..
```
</p>
</details>

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Cargo fmt all code
+c11ef36365f58f0b107c578ec9219ee1c86b09d2
```
Comment (taiki-e) [inline .git-blame-ignore-revs:2]: We usually merge PR by rebase or squash, so the commits merged into master will be different commit. Could you separate this into another PR?

File: .rustfmt.toml [hunk 2]
```diff
@@ -1 +1,2 @@
-disable_all_formatting = true
+#disable_all_formatting = true
```
Comment (taiki-e) [inline .rustfmt.toml:1]:
```suggestion
```

**Review by taiki-e — COMMENTED** [review]
LGTM. r=me after nit addressed

File: .git-blame-ignore-revs (hunk 1 above)
Comment (exrook) [reply .git-blame-ignore-revs]: sure thing, I'll make the PR once this gets merged

**Comment by exrook** [comment]
Recreated on top of master and fixed .rustfmt.toml
r? @taiki-e

**Review by taiki-e — APPROVED** [review]
Thanks!

**Comment by ibraheemdev** [comment]
Why don't you add a `cargo fmt` action to the gh workflow?

**Comment by taiki-e** [comment]
Filed #2421 to add fmt check to CI
